Docs updates

braingram opened the pull request:
<!-- If this PR closes a GitHub issue, reference it here by its number -->
Closes https://github.com/spacetelescope/roman_datamodels/issues/583

<!-- describe the changes comprising this PR here -->
Removes out-dated sections of the documentation. We may want to update rather than delete some of this content but I question how much of the deleted content was helpful to users. I'll leave some inline comments.

Overall I think the current documentation is too developer focused.  If we largely rethink the documentation of this package (after the intermediate release) and make it more user focused I think we will end up with a more helpful documentation. I will open a ticket to discuss and track that work. https://github.com/spacetelescope/roman_datamodels/issues/594

<!-- if you can't perform these tasks due to permissions, please ask a maintainer to do them -->
## Tasks
- [ ] Update or add relevant `roman_datamodels` tests.
- [ ] Update relevant docstrings and / or `docs/` page.
- [ ] Does this PR change any API used downstream? (If not, label with `no-changelog-entry-needed`.)
  - [ ] Write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types).
  - [ ] Start a `romancal` regression test (https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) with this branch installed (`"git+https://github.com/<fork>/rad@<branch>"`).

<details><summary>News fragment change types:</summary>

- ``changes/<PR#>.feature.rst``: new feature
- ``changes/<PR#>.bugfix.rst``: fixes an issue
- ``changes/<PR#>.doc.rst``: documentation change
- ``changes/<PR#>.removal.rst``: deprecation or removal of public API
- ``changes/<PR#>.misc.rst``: infrastructure or miscellaneous change
</details

PaulHuwe reviewed (approved):
LGTM. Do you want to move unaddressed suggestions into a new issue for the future?

braingram commented:
> LGTM. Do you want to move unaddressed suggestions into a new issue for the future?

Thanks. I'll update https://github.com/spacetelescope/roman_datamodels/issues/594
